From 1439b856531fdc0a18fd04ae0501f55c1f63048b Mon Sep 17 00:00:00 2001
From: "Philip, Avinash" <avinashphilip@ti.com>
Date: Mon, 23 Jan 2012 13:01:30 +0530
Subject: [PATCH 0739/1092] arm:omap:am33xx - update GPMC hwmod data

SOFT_RESET flag is added in GPMC_SYSCONFIG flags in hwmod data.
Resetting the module causes kernel crash on gpmc_init().  To fix the
crash base address and size has been updated for AM33xx SOC.

Signed-off-by: Philip, Avinash <avinashphilip@ti.com>
---
 arch/arm/mach-omap2/gpmc.c                 | 5 +++++
 arch/arm/mach-omap2/omap_hwmod_33xx_data.c | 3 ++-
 2 files changed, 7 insertions(+), 1 deletion(-)

diff --git a/arch/arm/mach-omap2/gpmc.c b/arch/arm/mach-omap2/gpmc.c
index 1cc4544..f3c132c 100644
--- a/arch/arm/mach-omap2/gpmc.c
+++ b/arch/arm/mach-omap2/gpmc.c
@@ -341,6 +341,11 @@ static void gpmc_cs_get_memconf(int cs, u32 *base, u32 *size)
 	*base = (l & 0x3f) << GPMC_CHUNK_SHIFT;
 	mask = (l >> 8) & 0x0f;
 	*size = (1 << GPMC_SECTION_SHIFT) - (mask << GPMC_CHUNK_SHIFT);
+
+	if (cpu_is_am33xx()) {
+		*base = 0x8000000;
+		*size = 0x10000000;
+	}
 }
 
 static int gpmc_cs_mem_enabled(int cs)
diff --git a/arch/arm/mach-omap2/omap_hwmod_33xx_data.c b/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
index 3a5f3a3..9b15ffe 100644
--- a/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
+++ b/arch/arm/mach-omap2/omap_hwmod_33xx_data.c
@@ -925,7 +925,8 @@ static struct omap_hwmod_class_sysconfig gpmc_sysc = {
 	.sysc_offs	= 0x10,
 	.syss_offs	= 0x14,
 	.sysc_flags	= (SYSC_HAS_AUTOIDLE | SYSC_HAS_SIDLEMODE |
-				SYSC_HAS_MIDLEMODE),
+				SYSC_HAS_MIDLEMODE | SYSC_HAS_SOFTRESET |
+				SYSS_HAS_RESET_STATUS),
 	.idlemodes	= (SIDLE_FORCE | SIDLE_NO | SIDLE_SMART |
 				MSTANDBY_FORCE | MSTANDBY_NO | MSTANDBY_SMART),
 	.sysc_fields	= &omap_hwmod_sysc_type4,
-- 
1.7.11.2

